### PR TITLE
feat PUD-1153 (pci.storage.databases):  add input rules visualisation for add, fork and edit pages

### PR DESCRIPTION
**Pull request by jperchoc**
PUD-1153

Signed-off-by: Jonathan Perchoc <jonathan.perchoc@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/aiven-ga1-pud`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #PUD-1153
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description
Add input rules visualisation for those pages:
- add a cluster
- change cluster name from list
- change cluster name from cluster details
- fork

### CODE REVIEW

**Review by frenautvh — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js [hunk 1]
```diff
@@ -23,7 +27,9 @@ export default class {
     this.DatabaseService = DatabaseService;
     this.Poller = Poller;
     this.NAME_PATTERN = NAME_PATTERN;
+    this.MIN_NAME_LENGTH = MIN_NAME_LENGTH;
     this.MAX_NAME_LENGTH = MAX_NAME_LENGTH;
+    this.regexp = new RegExp(this.NAME_PATTERN);
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js:32]: Not already a regex ?

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import angular from 'angular';
+import '@ovh-ux/ui-kit';
+
+import component from './input-rule.component';
+
+const moduleName = 'ovhManagerPciStoragesDatabasesInputRule';
+
+angular
+  .module(moduleName, ['oui'])
+  .component('inputRule', component)
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/index.js:10]: Name is a bit generic for a scoped component

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.controller.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export default class PciInputRuleController {}
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.controller.js:1]: Needed ?

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<span class="oui-password-rule">
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.html:1]: a `oui-field` or `oui-password` doesn't suit the needs ?

**Review by marie-j — CHANGES_REQUESTED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/add/add.controller.js:32]: You're right, I'll change this to use the pattern directly

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/index.js (hunk 2 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/index.js:10]: Would `pciInputRule` be better name to have it scoped to the pci universe ?

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.controller.js (hunk 3 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.controller.js:1]: Not anymore, good catch !

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.html (hunk 4 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.html:1]: `oui-password` hides the text, so it's not possible to use it here.
`oui-field` works if there is only one rule to check, but here we want to check more than one

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/index.js (hunk 2 above)
Comment (marie-j) [reply packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/index.js:10]: Maybe still too generic as we are in /storages/databases, something like `pciStorageDatabaseInputRule`

Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/index.js]: Changed here: https://github.com/ovh/manager/pull/6263/commits/c1d3c5629b9225d2f11784fb88688fc9b4389a2d

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.controller.js (hunk 3 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/components/input-rule/input-rule.controller.js]: Changed here: https://github.com/ovh/manager/pull/6263/commits/c1d3c5629b9225d2f11784fb88688fc9b4389a2d

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.constants.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-export const NAME_PATTERN = /^([a-zA-Z0-9_.-]{3,40})$/;
+export const NAME_PATTERN = /^([a-zA-Z0-9_.-]*)$/;
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/storages/databases/add/add.constants.js:1]:
```suggestion
export const NAME_PATTERN = /^([\w.-]*)$/;
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.constants.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 export const ADD_USER_FORM_RULES = {
   name: {
-    pattern: /^[a-zA-Z0-9_][a-zA-Z0-9_.-]{0,31}$/,
+    pattern: /^[a-zA-Z0-9_][a-zA-Z0-9_.-]{0,}$/,
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.constants.js:3]: For simplicity
```suggestion
    pattern: /^\w[\w.-]*$/,
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/add/add.constants.js (hunk 5 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/add/add.constants.js]: Changed here: https://github.com/ovh/manager/pull/6263/commits/333eeca5e16d628aaf730ba49688c30041b03112

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.constants.js (hunk 6 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.constants.js]: Changed here: https://github.com/ovh/manager/pull/6263/commits/333eeca5e16d628aaf730ba49688c30041b03112

**Review by JeremyDec — APPROVED** [review]

**Pull request merged**